sqlglot/dialects/tsql.py

```diff
@@ -539,3 +615,35 @@ def returning_sql(self, expression: exp.Returning) -> str:
             into = self.sql(expression, "into")
             into = self.seg(f"INTO {into}") if into else ""
             return f"{self.seg('OUTPUT')} {self.expressions(expression, flat=True)}{into}"
+
+        def transaction_sql(self, expression: exp.Transaction) -> str:
+            this = self.sql(expression, "this")
+            this = f" {this}" if this else ""
+            mark = expression.args.get("mark")
+            mark = f" WITH MARK {mark}" if mark else ""
```

Comment:
This is needed otherwise you implicitly use the default dialect for `mark` when interpolating it.

```suggestion
            mark = self.sql(expression, "mark")
            mark = f" WITH MARK {mark}" if mark else ""
```

Reply:
Done